src/components/file-upload/index.md.njk

```diff
@@ -23,6 +23,54 @@ There are 2 ways to use the file upload component. You can use HTML or, if you
 
 {{ example({group: "components", item: "file-upload", example: "default", html: true, nunjucks: true, open: false, id: "default-2"}) }}
 
+### Error messages
+
+Check that a valid date is entered by the user. Invalid dates should be reported to the user like this:
+
+{{ example({group: "components", item: "file-upload", example: "error", html: true, nunjucks: true, open: false, size: "m"}) }}
+
+Make sure errors follow the guidance in [error message](/components/error-message/) and have specific error messages for specific error states.
+
+#### If nothing has been uploaded
+Say ‘Select a [whatever they need to upload]’.<br>
+For example, ‘Select a report’.
+
+#### If the file is the wrong file type
+Say‘The [whatever] must be a [list of file types]’.<br>
+For example, ‘The report must be a CSV or ODS’ or ‘The upload must be a JPG, BMP, PNG, TIF or PDF’.
```

Comment:
I'm not sure how well 'upload' works as a noun here – how about

```suggestion
For example, ‘The report must be a CSV or ODS’ or ‘The uploaded file must be a JPG, BMP, PNG, TIF or PDF’.
```

?

Reply:
Actually, Would using 'selected file' work better throughout?

- ‘The selected file must be a CSV or ODS’ or ‘The selected file must be a JPG, BMP, PNG, TIF or PDF’.
- ‘The selected file must be smaller than 2MB’
- ‘The selected file is empty – select a CSV with data in it’
- ‘The selected file contains a virus’
- ‘The selected file is password protected’

Reply:
'uploaded file' is better than 'upload' and I will stick with upload rather than select.

Reply:
Using 'selected file' for all messages is a good idea. It means they will be less specific than using the file type or description of what the file is.

Maybe a couple of examples would make sense with selected file and with something more specific. I have added this to the latest commit.

Reply:
If possible, can we just exclusively use selected file? I think it works fine, and encouraging people to switch between naming the thing (eg, 'the report') and using 'selected file' seems like it could be problematic.

Reply:
Agreed.